deny.toml

```diff
@@ -31,6 +31,8 @@ allow = [
     # Pulled in via aws_lc_rs when using rustls-tls and aws-lc-rs features
     # https://openssl-library.org/source/license/index.html
     "OpenSSL",
+    # Pulled in via hyper-rustls when using the webpki-roots feature
+    "MPL-2.0",
```

Comment:
I think works, but it's better to have a specific exception with licenses.clarify below rather than doing a blanket allow.

Reply:
Fair point, I'll add one :)

Reply:
Since MPL-2.0 (unlike e.g. the ring license) is a well-defined license cargo-deny seems to be aware of, I figured it would make sense to add it as an exception entry rather than a clarify entry, which would attempt to read the license file from the repo like in ring's case. Does that make sense, or am I missing something about how this is supposed to work?

Reply:
It does seem to be well-defined, but it also does seem to [come with restrictions](https://opensource.stackexchange.com/questions/8831/pros-and-cons-of-using-mpl-2-0-license) that not everyone may want.

To me it sounds nicer to have it codified as one of the exceptions that people opt-in to by using the feature, rather than something that people have to scan for. Of course, people kind of have to scan their dependency tree for licenses transitively for these things regardless, but I think it's useful to know when a new dependency introduces constraints, and this kind of bypasses it a bit.

Maybe I am reading too much into it though.

Reply:
> It does seem to be well-defined, but it also does seem to [come with restrictions](https://opensource.stackexchange.com/questions/8831/pros-and-cons-of-using-mpl-2-0-license) that not everyone may want.

Agree, definitely!

My understanding from the cargo-deny docs was just that `LicenseRef-` + `clarify` was for licenses that aren't on the SPDX list (e.g. custom ones, like the ring one) that have to be fetched from a repo's source code and `exceptions` + `allow` for those on the SPDX list that can just be referenced by their canonical identifier without having to fetch them from some repo (that's what I mainly meant by well-defined).

But I'm new to cargo-deny and you obviously know the conventions (as well as your preferences for the project) way better than I do, I might very well be wrong here. So your preferred way of handling the MPL in this case would be creating a `"LicenseRef-MPL-2.0-webpki-roots"` entry on the `allow` list and a corresponding `[[licenses.clarify]]` entry referencing the license file on the crate's source code?

Reply:
Ahhh, my apologies! You are right indeed.

I had conflated the the `clarifications` with `exceptions` list. Given that `exceptions` is indeed scoped to specific deps rather than working like the global `allow` list, it seems to me like the right place for it. Sorry long time between looking at this stuff.

Reply:
No worries at all, cargo-deny confused me too. Thanks for being so responsive! :) And thanks again for the great work you're doing here!

Happy to see this small contribution land!